api/src/controllers/login.js

```diff
@@ -115,7 +115,12 @@ const setLocaleCookie = (res, locale) => {
   res.cookie('locale', locale, options);
 };
 
-const getRedirectUrl = userCtx => {
+const getRedirectUrl = (userCtx, req) => {
+  if (req.headers.referer){
+    const url = new URL(req.headers.referer);
```

Comment:
Be careful here with where we are redirecting. https://owasp.org/www-project-cheat-sheets/cheatsheets/Unvalidated_Redirects_and_Forwards_Cheat_Sheet.html

Reply:
Valid concern, we shouldn't blindly trust headers from the client. If the redirect points to the same origin then we should allow it.